src/test/resources/draft2020-12/issue495.json

```diff
@@ -1,61 +1,105 @@
 [
     {
         "description": "issue495 using ECMA-262",
-        "regex": "ecma-262",
         "schema": {
             "$schema": "https://json-schema.org/draft/2020-12/schema",
-            "pattern": "^[a-z]{1,10}$",
+            "patternProperties": {
+              "^[a-z]{1,10}$": true,
+              "(^1$)": true
+            },
             "unevaluatedProperties": false
         },
         "tests": [
             {
                 "description": "an expected property name",
+                "regex": "ecma-262",
                 "data": { "aaa": 3 },
                 "valid": true
             },
+            {
+                "description": "another expected property name",
+                "regex": "jdk",
+                "data": { "1": 3 },
+                "valid": true
+            },
             {
                 "description": "trailing newline",
+                "regex": "ecma-262",
                 "data": { "aaa\n": 3 },
-                "valid": false
+                "valid": false,
+                "disabled": true,
+                "comment": "Test fails"
+            },
+            {
+                "description": "another trailing newline",
+                "regex": "jdk",
+                "data": { "1\n": 3 },
+                "valid": false,
+                "disabled": true,
+                "comment": "Test fails"
             },
             {
                 "description": "embedded newline",
+                "regex": "ecma-262",
                 "data": { "aaa\nbbb": 3 },
                 "valid": false
             },
             {
                 "description": "leading newline",
+                "regex": "ecma-262",
                 "data": { "\nbbb": 3 },
                 "valid": false
             }
         ]
     },
     {
         "description": "issue495 using Java Pattern",
-        "regex": "jdk",
         "schema": {
             "$schema": "https://json-schema.org/draft/2020-12/schema",
-            "pattern": "^[a-z]{1,10}$",
+            "patternProperties": {
+              "^[a-z]{1,10}$": true,
+              "(^1$)": true
+            },
             "unevaluatedProperties": false
         },
         "tests": [
             {
                 "description": "an expected property name",
+                "regex": "jdk",
                 "data": { "aaa": 3 },
                 "valid": true
             },
+            {
+                "description": "another expected property name",
+                "regex": "jdk",
+                "data": { "1": 3 },
+                "valid": true
+            },
             {
                 "description": "trailing newline",
+                "regex": "jdk",
                 "data": { "aaa\n": 3 },
-                "valid": false
+                "valid": false,
+                "disabled": true,
+                "comment": "Test fails"
+            },
```

Comment:
This is the only test case on which this PR regresses. Both regex implementation suffer from the same issue:
```
jshell> Pattern.compile("^a$").matcher("a\n").find()
$1 ==> true
jshell> Pattern.compile("(^a$)").matcher("a\n").find()
$2 ==> true
```

Reply:
Okay, I found why this happens. [The documentation states](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/regex/Pattern.html#lt):
>  The regular expression `$` matches at the end of the entire input sequence, but also matches just before the last line terminator if this is not followed by any other input character. Other line terminators are ignored, including the last one if it is followed by other input characters.

So _in theory_ we could avoid this by transforming all `$` anchors by prepending a negative lookahead, like so:
```
jshell> Pattern.compile("^a(?!\n|\r\n|\r|\\u0085|\\u2028|\\u2029)$").matcher("a\n").find()
$1 ==> false
```

But that will require careful parsing of the input regex, because the `$` could be escaped, or inside a `\Q`...`\E` section, or... Quite frankly, that way lies madness; would like to keep it out of this PR.